### DISCUSSION TITLE
More suggestive error for uninitialized parameters

### DISCUSSION THREAD

**Discussion by rolandgvc** [google/flax#558] [Ideas] [answered]
Hi guys! While playing with the Linen API for transfer learning, I've found the `AssertionError: Need PRNG for "params"` error quite confusing when trying to match the parameter namings to the model's layers. I was wondering if we could have a more explicit error when certain variables haven't been (or cannot be) initialized.

**Comment by avital** [comment, marked as answer]
Hi Roland! I filed https://github.com/google/flax/issues/489, does this cover the situation that you've encountered?

**Reply by rolandgvc** [reply]
Hi Avital, yes, that's great! Thank you.
